tests/upgrade/rules/test_airflow_macro_plugin_removed.py

```diff
@@ -0,0 +1,70 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from contextlib import contextmanager
+from unittest import TestCase
+
+from tempfile import NamedTemporaryFile
+from tests.compat import mock
+
+from airflow.upgrade.rules.airflow_macro_plugin_removed import AirflowMacroPluginRemovedRule
+
+
+@contextmanager
+def create_temp_file(mock_list_files, lines):
+    temp_file = NamedTemporaryFile("w+")
+    mock_list_files.return_value = [temp_file.name]
+    for line in lines:
+        temp_file.write(line)
+        temp_file.write("\n")
+    temp_file.flush()
+    yield temp_file
```

Comment:
I'm afraid that the temp file is not deleted (but I'm not sure), how about:
```suggestion
    with NamedTemporaryFile("w+") as temp.file:
        mock_list_files.return_value = [temp_file.name]
        temp_file.writelines(lines)
        temp_file.flush()
        yield temp_file
```